core/src/com/unciv/ui/utils/UnitGroup.kt

```diff
@@ -72,44 +169,39 @@ class UnitGroup(val unit: MapUnit, val size: Float): Group() {
     fun selectUnit() {
 
         //Make unit icon background colors fully opaque when units are selected
-        background?.color?.a = 1f
+        flagBg?.children?.forEach { it.color?.a = 1f }
 
-        //If unit is idle, leave unitBaseImage and actionGroup at 50% opacity when selected
+        //If unit is idle, leave actionGroup at 50% opacity when selected
         if (!unit.isIdle()) {
-            unitBaseImage.color.a = 0.5f
             actionGroup?.color?.a = 0.5f
         } else { //Else set to 100% opacity when selected
-            unitBaseImage.color.a = 1f
             actionGroup?.color?.a = 1f
         }
 
-        val whiteHalo = getBackgroundImageForUnit()
-        val whiteHaloSize = 30f
-        whiteHalo.setSize(whiteHaloSize, whiteHaloSize)
-        whiteHalo.center(this)
-        addActor(whiteHalo)
-        whiteHalo.toBack()
+        // Unit base icon is faded out only if out of moves
+        if (unit.currentMovement == 0f && !unit.canAttack()) {
+            unitBaseImage.color.a = 0.5f
+            flagHalo?.color?.a = 0f
+            flagBg?.children?.forEach { it.color.a = 0.5f }
+        } else {
+            unitBaseImage.color.a = 1f
+            flagHalo?.color?.a = 1f
+            flagBg?.children?.forEach { it.color.a = 1f }
+        }
 
+        flagSelection?.apply { color.a = 1f }
+        flagHalo?.apply { color.a = 1f }
```

Comment:
```suggestion
        flagSelection.color.a = 1f
        flagHalo.color.a = 1f
```
Why all the complication